### PR TITLE
Bug fix and adding lseek syscall tests

### PR DESCRIPTION
**Pull request by Owlz**
Getting my feet wet in writing some test cases. lseek was an easy option for that.

Also, while testing, i found that the implementation of posix.seek subtracted from end instead of adding. Also, since GNU whence is a feasible option, i changed the logic to output an error if whence was not 3. While not likely, it's certainly possible and should be accounted for.

### CODE REVIEW

**Comment by Owlz** [comment]
For what it's worth, i find the lint check for spacing to cause the code to be less readable. It does not allow for ASCII headings inside blocks of code, such as:

```python
# 
# Subheading 1
#
```

I've found having ASCII comment headings inside code to help make it more readable.

**Comment by rhelmot** [comment]
I think you're misunderstanding the lint error. You're allowed to have blank lines, and you're allowed to have block comments like that, in any combination, but you're not allowed to have any line that ends with a whilespace character. Basically, your file shouldn't match `[ \t]+$`.

**Comment by Owlz** [comment]
Oh. Heh. Gotcha.

File: simuvex/plugins/posix.py [hunk 1]
```diff
@@ -282,16 +282,19 @@ def seek(self, fd, seek, whence):
...
             if size is None:
                 # The file is not seekable
                 l.error('File with fd %d is not seekable.', fd)
                 return -1
-            f.seek(size - seek)
+            f.seek(size + seek)
             return 0
 
+        else:
+            l.error("Unknown whence for file seek of \"%s\". Note, GNU extensions are not supported right now.",whence)
```
Comment (rhelmot) [inline simuvex/plugins/posix.py]: add a `return -1` here

Comment (Owlz) [inline simuvex/plugins/posix.py]: Oops... Fixed.

**Comment by Owlz** [comment]
What are your guys' take on added unit tests for error cases? I.e.: in lseek I have an error set for symbolic whence and symbolic seek. Is it worthwhile to test for those cases? Would mean if that functionality is ever added the unit test would need to be updated as well.

**Comment by rhelmot** [comment]
More test cases are always good! If we refactor it, the test cases serve as a reminder that we're changing the intended behavior of the module, and should update the docs and tell people about it.

**Comment by Owlz** [comment]
Boom. 100% for lseek.

**Comment by Owlz** [comment]
lemme know if you need more changes.

**Comment by zardus** [comment]
Sorry, we're rushing toward yet another deadline, so we've been falling behind :-)

**Pull request merged**